### DISCUSSION TITLE
"Unknown Epoch" with reliable streams?

### DISCUSSION THREAD

**Discussion by maxwell** [anycable/anycable#217] [Q&A] [answered]
HI,

I noticed I am frequently getting this error in my logs for my anycable server (1.4.1). I am using Speedy Streams and Reliable Streams.  I'm not sure if this is a bug or just helpful message, but it seems to be happening a few times a minute so not sure if it means i have something misconfigured.

```
 W 2023-10-10T16:38:39.568Z context=node sid=ca432edb-2779-4cfa-84e8-8bae88f982e1 Failed to handle incoming message '{"identifier":"{\"channel\":\"Turbo::StreamsChannel\",\"signed_stream_name\":\"IloybGtPaTh2WW1GamEyVnlhMmwwTDBOeWIzZGtablZ1WkdsdVp6bzZVSEp2YW1WamRGTjBZWFF2TlRJdyI=--0529b0252713bbdc9ac330de50298085a9c3835467056b6772fa1b78227d36a1\"}","command":"history","history":{"since":1696955902,"streams":{"Z2lkOi8vYmFja2Vya2l0L0Nyb3dkZnVuZGluZzo6UHJvamVjdFN0YXQvNTIw":{"epoch":"MSmU","offset":17}}}}' with error: Unknown epoch: MSmU, current: 45tx

```

**Comment by palkan** [comment, marked as answer]
Hey,

It's not a bug just a warning telling you that history request couldn't be fulfilled due to client and server states being inconsistent. That may happen if you use an in-memory store for streams history and restart your server.

**Reply by maxwell** [reply]
Got it, thank you for the quick response.  This might be a great place to plug anycable-go pro in the app..right? if I want multi node, thats the reason to upgrade?

**Reply by palkan** [reply]
Correct. Currently, multi-node, persistent store for reliable streams is only available in Pro.
